sql/sql_parse.cc

```diff
@@ -2168,6 +2182,10 @@ bool dispatch_command(THD *thd, const COM_DATA *com_data,
   mysql_audit_notify(thd, AUDIT_EVENT(MYSQL_AUDIT_COMMAND_END), command,
                      command_name[command].str);
 
+  if (command == COM_QUERY && !thd->trace_id.empty()) {
+    thd->pc_val = thd->query_perf->getAndStop();
+  }
+
```

Comment:
Might useful to DBUG_ASSERT that !thd->trace_id.empty() implies command == COM_QUERY.

Reply:
done